smash/core/simulation/_doc.py

```diff
@@ -865,12 +865,32 @@ def _gen_docstring_from_base_doc(
         nindent=1,
     )
     + """
+callback : callable or None, default None
+    A callable called after each iteration with the signature ``callback(iopt: Optimize)``, where
+    ``iopt`` is a keyword argument representing an instance of the `Optimize` class that contains
+    intermediate optimization results with attributes:
+
+    - ``'control_vector'``: The current control vector.
+    - ``'cost'``: The current cost value.
+    - ``'projg'``: The projected gradient, available if using gradient-based optimizers.
+    - ``'net'``: The regionalization neural network state, available if using ``'ann'`` **mapping**.
+
+    >>> iter_cost = []  # to get the cost values through iterations
+    >>> def callback_cost(iopt, icost=iter_cost):
+    ...     icost.append(iopt.cost)
+    >>> callback = callback_cost
```

Comment:
Je comprends pas bien la ligne:
``` python
callback = callback_cost
```

Reply:
oui pas bien compris ce que tu fais au dessus, append de valeurs de cost sur les ités ? clarifier le commentaire de l878